src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamespaceSymbol.cs

```diff
@@ -89,29 +91,42 @@ private ImmutableArray<NamedTypeSymbol> GetMemberTypesPrivate()
 
         public sealed override ImmutableArray<Symbol> GetMembers(ReadOnlyMemory<char> name)
         {
+            var (lastName, lastResult) = _lastGetMembersResult;
+            if (lastName.Span.SequenceEqual(name.Span))
+            {
+                return lastResult;
+            }
+
             EnsureAllMembersLoaded();
 
-            PENestedNamespaceSymbol ns = null;
             ImmutableArray<PENamedTypeSymbol> t;
 
-            if (lazyNamespaces.TryGetValue(name, out ns))
+            if (lazyNamespaces.TryGetValue(name, out PENestedNamespaceSymbol ns))
             {
                 if (lazyTypes.TryGetValue(name, out t))
                 {
                     // TODO - Eliminate the copy by storing all members and type members instead of non-type and type members?
```

Comment:
I see this TODO suggests a method of avoiding the copy, should we pursue this instead?

Reply:
Probably. @sharwell what do you think?

Reply:
No, this is an uncommon case (a namespace, at least in C#, cannot have a type and a nested namespace with the same name).